src/graphql/graphql-api.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * GraphQL API
+ *
+ * @package vip-block-data-api
+ */
+
+namespace WPCOMVIP\BlockDataApi;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * GraphQL API to offer an alternative to the REST API.
+ */
+class GraphQLApi {
+	/**
+	 * Initiatilize the graphQL API, if its allowed
+	 *
+	 * @access private
+	 */
+	public static function init() {
+		$is_graphql_to_be_enabled = apply_filters( 'vip_block_data_api__is_graphql_enabled', true );
+
+		if ( ! $is_graphql_to_be_enabled ) {
+			return;
+		}
+
+		add_filter( 'vip_block_data_api__sourced_block_result_transform', [ __CLASS__, 'transform_block_attributes' ], 10, 5 );
+
+		add_action( 'graphql_register_types', [ __CLASS__, 'register_types' ] );
+	}
+
+	/**
+	 * Extract the blocks data for a post, and return back in the format expected by the graphQL API.
+	 *
+	 * @param  WPGraphQL\Model\Post $post_model Post model for post.
+	 * @return array
+	 */
+	public static function get_blocks_data( $post_model ) {
+		$post_id        = $post_model->ID;
+		$post           = get_post( $post_id );
+		$filter_options = [ 'graphQL' => true ];
```

Comment:
Using this as a way to signify that it's a graphQL request and so the filter would only target those calls.